src/dataset/tag.rs

```diff
@@ -0,0 +1,135 @@
+use std::fmt;
+
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Deserialize, Serialize)]
+pub enum Tag {
+    Other(String),
+    Wrrl,
+    HwrmRl,
+    MsrRl,
+    BgRl,
+}
+
+impl From<String> for Tag {
+    fn from(val: String) -> Self {
+        Self::Other(val)
+    }
+}
+
+impl From<&'_ str> for Tag {
```

Comment:
I'll need some explanation here...

Reply:
The idea is to make it easy to construct `Tag` value from text, i.e. instead of doing

```rust
Tag::Other("foobar.to_owned())
```

one can do

```rust
Tag::from("foobar")
```

or even

```rust
"foobar".into()
```

if the type `Tag` can be inferred by the compiler.

The standard way to provide such lossless conversions is the [`From`](https://doc.rust-lang.org/stable/std/convert/trait.From.html) trait. In this particular case, I enable the above code snippets by providing conversions from `String` (owned UTF-8 encoded text) and `&str` (reference to UTF-8 text stored elsewhere).

The implementation for `&str` basically turns it into `String` using [`.to_owned()`](https://doc.rust-lang.org/stable/std/borrow/trait.ToOwned.html#tymethod.to_owned) and then relies on the implementation for `String` to do the actual conversion. (The same pattern is used in `license.rs` where we have `impl From<String> for License` and `impl From<Option<String>> for License` where the latter is implemented using the former.)

Reply:
(If the `&'_ str` is the issue, this is just shorthand for

```rust
impl<'a> From<&'a str> for Tag {
```

meaning that a lifetime that is present but that the impl does not care about can be elided as `_`.)